addons/mass_mailing/models/mass_mailing.py

```diff
@@ -341,15 +356,9 @@ def get_recipients(self, cr, uid, ids, model=None, context=None):
         return res
 
     def on_change_campaign_name(self, cr, uid, ids, name, context=None):
-        if name:
-            mass_mailing_campaign = self.browse(cr, uid, ids, context=context)
-            if mass_mailing_campaign.campaign_id:
-                utm_campaign_id = mass_mailing_campaign.campaign_id.id
-                self.pool['utm.campaign'].write(cr, uid, [utm_campaign_id], {'name': name}, context=context)
-            else:
-                utm_campaign_id = self.pool['utm.campaign'].create(cr, uid, {'name': name}, context=context)
-
-            return {'value': {'campaign_id': utm_campaign_id}}
+        if self.browse(cr, uid, ids, context=context).campaign_id.id is False:
```

Comment:
The issue is that I think you could have a void ids (when creating a new record for example). To check.

Can't you keep a void onchange ? Why linking to the first found campaign, if we create a new utm campaign at each mailing campaign ?